### PR TITLE
Added FlightService

### PR DESCRIPTION
**Pull request by jirismika**
Hello, I've implemented the FlightService as was previously noted in #33 

Options are extended for a required key `key`. Example:
```
options = {
  airline: 'OS',
  flightNumber: '703',
  departure: '2016-11-21',
  key: 'MQ==',
};
```

### CODE REVIEW

File: src/Services/Air/AirErrors.js [hunk 1]
```diff
@@ -61,15 +77,22 @@ Object.assign(AirRuntimeError, createErrorsList({
   NoValidFare: 'No valid fare for input criteria.',
   TravelersListError: 'Not all BookingTravelers present in list or wrong lookup keys provided',
 }, AirRuntimeError));
+Object.assign(AirFlightInfoRuntimeError, createErrorsList({
+  FlightNotFound: 'Flight not found',
+  AirlineNotSupported: 'Airline not supported',
+  InvalidFlightNumber: 'Invalid flight number',
+}, AirRuntimeError));
```
Comment (dchertousov) [inline src/Services/Air/AirErrors.js]: Parent of the errors has to be `AirFlightInfoRuntimeError`, not `AirRuntimeError`

**Comment by shmuga** [comment]
Everything looks good to me except new service creation.
It's probably better to put this function to `AirService`.

**Comment by dchertousov** [comment]
@jirismika , do you find it possible to make `flightInfo` function a part of `AirService`?

**Comment by jirismika** [comment]
@dchertousov If you mean to do something like 

```
const airService = uapi.createAirService(settings);
airService.flightInfo(options);
```
... than yes, I do. Is this the final decision to implement it as part of AirService?

**Comment by Smotrov** [comment]
What does this `key` parameter for?

**Comment by dchertousov** [comment]
@jirismika , yes, we see it as a simplification to divide all requests into exactly four types:
- Air
- Hotel
- Utils
- Terminal

**Comment by jirismika** [comment]
Okay, I'm going to move it to AirService.

Key is (to simplify it) a "grouping" parameter :
[The base-64 encoded UUID identifier that links the Flight Information responses to the request criteria. The value that is sent is returned in the resulting FlightInfo(s) elements.](https://support.travelport.com/webhelp/uapi/uAPI.htm#../Subsystems/Schemas/Content/Schemas/FlightInformationReq.html)

**Comment by dchertousov** [comment]
@jirismika , how does the user know which key to send for which request?

**Comment by Smotrov** [comment]
@jirismika sorry, but I can't find any information about "grouping" parameter in provided links.

**Comment by shmuga** [comment]
@jirismika I've looked into WSDL and found that you can pass `<!--1 or more repetitions:-->`  of `FlightInfoCriteria` to this request. Maybe it should be better to pass an array of such options as you mentioned?

Example:
```
[{
  airline: 'OS',
  flightNumber: '703',
  departure: '2016-11-21',
}, {
  airline: 'PS',
  flightNumber: '748',
  departure: '2016-11-21',
} ]
```

And in this case, you can set the `key` property as element index. 
This [line](https://github.com/Travelport-Ukraine/uapi-json/blob/master/src/Services/Air/templates/AIR_PRICE_REQ.xml#L25) should help you with that.

And please provide `examples` in the relevant folder for this request. Thanks.

**Comment by jirismika** [comment]
Well, you've actually made me think about it again and I didn't realized earlier, that flightInfo is not returning this key back ... So I also have to extend the response. And as I tested it now, the key can be an empty string, so do you want me to do that parameter optional?

@dchertousov It depends on the user to define the key.

@smotrov 
FlightInformationReq[`key`](https://support.travelport.com/webhelp/uapi/uAPI.htm#../Subsystems/Schemas/Content/Schemas/FlightInformationReq.html)
> The base-64 encoded UUID identifier that links the Flight Information responses to the request criteria. The value that is sent is returned in the resulting FlightInfo(s) elements.

FlightInformationRsp[`CriteriaKey` ](https://support.travelport.com/webhelp/uapi/uAPI.htm#../Subsystems/Schemas/Content/Schemas/FlightInformationRsp.html)
>  	An identifier to link the flightinfo responses to the criteria in request. The value populated here is passed in request.

@shmuga Thanks, you were faster than me. Yes, the `key` is used always in multiple items in response to group/match/link/find the item on request side. Array of requests is not implemented yet, thanks for finding that. Added to my todo list:

TL;DR TODO: 
- question (for all): should be the `key` optional?
- array of requests (and responses)
- examples

**Comment by Smotrov** [comment]
@jirismika I doubt that we should put `key` to interface at all. Even as optional parameter.

**Comment by jirismika** [comment]
Ok, let me ask the question differently: 
Should the `key` be optional or we don't want to return the `key`(s) at all?

**Comment by Smotrov** [comment]
I think we should not return `key` at all.

File: src/Services/Air/templates/AIR_FLIGHT_INFORMATION_REQUEST.xml [hunk 2]
```diff
@@ -6,12 +6,14 @@
     <soapenv:Body>
         <air:FlightInformationReq TargetBranch="{{TargetBranch}}">
             <com:BillingPointOfSaleInfo OriginApplication="uAPI"/>
+            {{#flightInfoCriteria}}
             <air:FlightInfoCriteria
                 Carrier="{{airline}}"
                 DepartureDate="{{departure}}"
                 FlightNumber="{{flightNumber}}"
-                Key="{{key}}"
+                Key=""
```
Comment (dchertousov) [inline src/Services/Air/templates/AIR_FLIGHT_INFORMATION_REQUEST.xml]: Does it work with empty key?
Doesn't this make results mixed up?

File: docs/Air.md [hunk 3]
```diff
@@ -177,3 +178,20 @@ If the Universal Record locator code is known, a Universal Record can be retriev
...
+
+**Returns**: <code>Promise</code>
+**See**: [Flight Information](https://support.travelport.com/webhelp/uapi/uAPI.htm#Air/Flight_Information/Flight_Information.htm)
+
+| Param | Type | Description |
+| --- | --- | --- |
+| airline | <code>string</code> | The marketing carrier for the segment. |
+| departure | <code>string</code> | The departure date; either the current date or a date in the future, not include the time zone which is derived from the origin location. |
+| flightNumber | <code>string</code> | The flight number for the segment |
+| key | <code>string</code> | The base-64 encoded UUID identifier that links the Flight Information responses to the request criteria. The value that is sent is returned in the resulting FlightInfo(s) elements. |
```
Comment (dchertousov) [inline docs/Air.md]: I think `key` should be removed from the documentation.

File: src/Services/Air/templates/AIR_FLIGHT_INFORMATION_REQUEST.xml (hunk 2 above)
Comment (jirismika) [inline src/Services/Air/templates/AIR_FLIGHT_INFORMATION_REQUEST.xml]: Empty key works fine, there is a new `key` in response, but we are not returning the `key` at all.

**Comment by dchertousov** [comment]
Thank you for your work.
